### ISSUE TITLE
Inconsistent treatment of NULL from subqueries

### ISSUE THREAD

**Issue by janetzki** [hyrise/hyrise#1526]
**Actual behavior**
```sql
SELECT MIN(c_custkey) FROM customer WHERE c_custkey < -1
```
correctly returns `NULL`.

```sql
SELECT MIN(c_custkey) FROM customer WHERE c_custkey < (
  SELECT MIN(c_custkey) FROM customer WHERE c_custkey < -1
)
```
If I nest the query in itself, it still returns `NULL`.

```sql
SELECT MIN(c_custkey) FROM customer WHERE c_custkey < NULL
```
If I substitute the subquery, it breaks the Hyrise console:
```
terminate called after throwing an instance of 'std::logic_error'
  what():  Cannot convert from NullValue to anything but string
Aborted
```

**Expected behavior**
The second and third query should have the same result.

**Comment by mrzzzrm** [comment]
Is that inner query here what you meant?

```sql
SELECT MIN(c_custkey) FROM customer WHERE c_custkey < (
  SELECT MIN(c_custkey) FROM customer WHERE c_custkey
)
```
?

Because I would not expect that to return NULL, `WHERE c_custkey` is probably treated as "TRUE" for all c_custkey != 0

**Comment by janetzki** [comment]
Sorry, I missed adding `< -1`.

**Comment by mrzzzrm** [comment]
This looks like it'll probably get the same fix as #1525 (AllTypeVariant cast in ColumnVsValueScan that shouldn't happen; ExpressionEvaluator doing the scan instead)
